### PR TITLE
adds scope of variable documentation

### LINKED ISSUE #1082: Clarify variables scope in docs
We should clarify the scope of the variables documented in `docs/content/stack/configuration-variables.md`. They can be project level, global or both.

### PR DESCRIPTION
**Pull request by shelane**
I'm making a wild guess about some of these.

Closes #1082

### CODE REVIEW

File: docs/content/stack/configuration-variables.md [hunk 1]
```diff
@@ -8,31 +8,31 @@ aliases:
 
 Configuration variables can be specified in the `docksal.env` file or by running the `fin config set` command (e.g., `fin config set --global PROJECT_INACTIVITY_TIMEOUT="0.5h"`). All variables that are changed from their defaults below will require, at minimum, a project restart `fin restart` so that they can take effect. Additional steps may be required.
```
Comment (achekulaev) [inline docs/content/stack/configuration-variables.md]:
```suggestion

There are 2 scopes of variables: global and project. Global scope variables are supposed to be set in `~/.docksal/docksal.env` while project scope variables should be set in `$PROJECT_ROOT/.docksal/docksal.env`.

```

File: docs/content/stack/configuration-variables.md [hunk 2]
```diff
@@ -110,166 +112,166 @@ Designates whether to use Docker through VirtualBox or Native Docker. On Linux t
...
 User to be used for `fin exec`.
 
-## Image Variables
+## Image Variables (project)
 
 The following variables are used to set images to a specified version. Using them will prevent the images from being udpated
 when Docksal is updated.
 
-### CLI_IMAGE
+### CLI_IMAGE (project) (project)
```
Comment (achekulaev) [inline docs/content/stack/configuration-variables.md]:
```suggestion
### CLI_IMAGE (project)
```

File: docs/content/stack/configuration-variables.md [hunk 3]
```diff
@@ -56,24 +58,24 @@ Can be used in conjunction with [SSH agent forwarding](https://developer.github.
...
 
 When set to `1`, this will prevent Docksal from installing and checking for updates. This is usually good in combination with `CI=true`.
 
-### DOCKER_VERSION_ALERT_SUPPRESS
+### DOCKER_VERSION_ALERT_SUPPRESS (global)
 
 When set, suppresses alert about the outdated Docksal version.
 
-### DOCKSAL_ENVIRONMENT 
+### DOCKSAL_ENVIRONMENT (global)
```
Comment (achekulaev) [inline docs/content/stack/configuration-variables.md]:
```suggestion
### DOCKSAL_ENVIRONMENT (project)
```

File: docs/content/stack/configuration-variables.md [hunk 4]
```diff
@@ -110,166 +112,166 @@ Designates whether to use Docker through VirtualBox or Native Docker. On Linux t
...
+### DOCKSAL_VHOST_PROXY_IP (global)
 
 Used to set the IP address for the Docksal reverse proxy to listen on. When `CI` variable is set to `true` this will be set to `0.0.0.0`.
 
-### DOCKSAL_DNS_DOMAIN
+### DOCKSAL_DNS_DOMAIN (global)
 
 `Default: docksal`
 
 This is the domain to use which is tacked on to the end of the projects url.
```
Comment (achekulaev) [inline docs/content/stack/configuration-variables.md]:
```suggestion
This is the domain name used for Docksal projects url, i.e. `http://$PROJECT_NAME.$DOCKSAL_DNS_DOMAIN`. Project named `myproject` will result in `http://myproject.docksal` url by default.
```

File: docs/content/stack/configuration-variables.md [hunk 5]
```diff
@@ -110,166 +112,166 @@ Designates whether to use Docker through VirtualBox or Native Docker. On Linux t
...
 
-### CLI_USER
+### CLI_USER (global or project)
 
 `Default:  docker`
 
 User to be used for `fin exec`.
 
-## Image Variables
+## Image Variables (project)
```
Comment (achekulaev) [inline docs/content/stack/configuration-variables.md]:
```suggestion
## Image Variables
```

File: docs/content/stack/configuration-variables.md [hunk 6]
```diff
@@ -18,27 +20,27 @@ Defines a relative path to the Document Root of the web server. Final path will
```
Comment (achekulaev) [inline docs/content/stack/configuration-variables.md]:
```suggestion

`Default: '/Users'`

```

File: docs/content/stack/configuration-variables.md [hunk 7]
```diff
@@ -18,27 +20,27 @@ Defines a relative path to the Document Root of the web server. Final path will
 
 The location of the folder on the host machine to mount to VirtualBox. See [file sharing](/core/file-sharing/) for more information.
```
Comment (achekulaev) [inline docs/content/stack/configuration-variables.md]:
```suggestion
**macOS only.** Sets the location of the folder on the host machine to mount to VirtualBox. See [file sharing](/core/file-sharing/) for more information.
```

File: docs/content/stack/configuration-variables.md [hunk 8]
```diff
@@ -16,29 +18,31 @@ Defines a relative path to the Document Root of the web server. Final path will
...
+### IMAGE_SSH_AGENT (global)
 
 `Default: docksal/ssh-agent:1.0`
 
 The Docker image to use for SSH Agent. This contains the keys that are shared among the projects.
 
-### IMAGE_VHOST_PROXY 
+### IMAGE_VHOST_PROXY (global)
 
 `Default: docksal/vhost-proxy:1.3`
```
Comment (achekulaev) [inline docs/content/stack/configuration-variables.md]:
```suggestion
`Default: docksal/vhost-proxy:1.5` 
```

File: docs/content/stack/configuration-variables.md [hunk 9]
```diff
@@ -16,29 +18,31 @@ Defines a relative path to the Document Root of the web server. Final path will
...
 
-The location of the folder on the host machine to mount to VirtualBox. See [file sharing](/core/file-sharing/) for more information.
+`Default: '/Users'`
+
+**macOS only.** Sets the location of the folder on the host machine to mount to VirtualBox. See [file sharing](/core/file-sharing/) for more information.
 
-### IMAGE_SSH_AGENT
+### IMAGE_SSH_AGENT (global)
 
 `Default: docksal/ssh-agent:1.0`
```
Comment (achekulaev) [inline docs/content/stack/configuration-variables.md]:
```suggestion
`Default: docksal/ssh-agent:1.3`
```

File: docs/content/stack/configuration-variables.md [hunk 10]
```diff
@@ -16,29 +18,31 @@ Defines a relative path to the Document Root of the web server. Final path will
...
+### IMAGE_VHOST_PROXY (global)
 
 `Default: docksal/vhost-proxy:1.3` 
 
 Docker image to use as the VHOST Proxy. This forwards all requests to the appropriate container based upon the hostname that is being requested.
 
-### IMAGE_DNS
+### IMAGE_DNS (global)
 
 `Default: docksal/dns:1.0`
```
Comment (achekulaev) [inline docs/content/stack/configuration-variables.md]:
```suggestion
`Default: docksal/dns:1.1`
```

File: docs/content/stack/configuration-variables.md [hunk 11]
```diff
@@ -101,19 +105,19 @@ on a build/sandbox server via `fin config set --global DOCKSAL_ENVIRONMENT=ci`.
...
+### DOCKSAL_STATS_OPTOUT (global)
 
 `Default: 0` 
 
 Allow for collecting of statistical usage of docksal. When set to `1` this will no longer send statistics.
 
-### DOCKER_NATIVE
+### DOCKER_NATIVE (global)
 
 `Default: 0`
```
Comment (achekulaev) [inline docs/content/stack/configuration-variables.md]:
```suggestion
`Default: 0 (on Linux: 1)`
```

File: docs/content/stack/configuration-variables.md [hunk 12]
```diff
@@ -122,147 +126,147 @@ Designates whether to use Docker through VirtualBox or Native Docker. On Linux t
 For VirtualBox set to `0`.
 For Docker Native set to `1`.
 
-### DOCKSAL_DNS_UPSTREAM
+### DOCKSAL_DNS_UPSTREAM (global)
 
 Override the default DNS server that Docksal uses. For environments where access to Google DNS server (`8.8.8.8`) is blocked, it should be set to the LAN DNS server. This is often true for VPN users or users behind a corporate firewall.
 
-### DOCKSAL_VHOST_PROXY_IP
+### DOCKSAL_VHOST_PROXY_IP (global)
```
Comment (achekulaev) [inline docs/content/stack/configuration-variables.md]:
```suggestion

`Default: 192.168.64.100`

```

File: docs/content/stack/configuration-variables.md [hunk 13]
```diff
@@ -122,147 +126,147 @@ Designates whether to use Docker through VirtualBox or Native Docker. On Linux t
 For VirtualBox set to `0`.
 For Docker Native set to `1`.
 
-### DOCKSAL_DNS_UPSTREAM
+### DOCKSAL_DNS_UPSTREAM (global)
```
Comment (achekulaev) [inline docs/content/stack/configuration-variables.md]:
```suggestion

`Default: autodetected with fallback to 8.8.8.8`

```

File: docs/content/stack/configuration-variables.md [hunk 14]
```diff
@@ -122,147 +126,147 @@ Designates whether to use Docker through VirtualBox or Native Docker. On Linux t
...
-### DOCKSAL_DNS_DOMAIN
+### DOCKSAL_DNS_DOMAIN (global)
 
 `Default: docksal`
 
-This is the domain to use which is tacked on to the end of the projects url.
+This is the domain name used for Docksal projects url, i.e. `http://$PROJECT_NAME.$DOCKSAL_DNS_DOMAIN`. Project named `myproject` will result in `http://myproject.docksal` url by default.
 
-### DOCKSAL_NO_DNS_RESOLVER
+### DOCKSAL_NO_DNS_RESOLVER (global)
```
Comment (achekulaev) [inline docs/content/stack/configuration-variables.md]:
```suggestion

`Default: ''`

```

**Comment by shelane** [comment]
I had a hard time trying to merge develop into the branch. So, I did a "manual" merge by saving changes, creating a new branch from develop, and putting the new changes back in (being aware of what was removed from this doc in the develop branch).

**Comment by achekulaev** [comment]
It works for intermediate step, but in the future I would like to do the following, split the doc into Global (going first) and Project (going after it) sections. Inside those sections variables should be ordered in the alphabetical order by name. Variables belonging to both should be mentioned in global with remark that it can go to project too.

**Comment by shelane** [comment]
I have time to do this now.

**Comment by achekulaev** [comment]
CI variables are very separate, they are a use case of their own. 

The only reason Image variables are separate because in alphabetical order they would be scattered throughout the Project level variables. We can have them as a subsection of Project variables.

**Comment by lmakarov** [comment]
Made some updates. Waiting for Netlify to rebuild the preview to validate (Netlify is currently experiencing some issues).

**Pull request merged**